src/components/Account-panel/Order-history/OrderProducts.tsx

```diff
@@ -0,0 +1,183 @@
+import { Avatar, Box, Typography, Divider } from '@mui/material'
+import { getProductLabel, SelectedOrder } from './OrdersList'
+import { Link } from 'react-router-dom'
+
+type OrderProductsProps = {
+  products: any[]
+  orderId: SelectedOrder | null
+}
+
+export const getProductGrams = (count: number) => {
+  if (count === 1) return 'штука'
+  if (count >= 2 && count <= 4) return 'штукі'
+  return 'штук'
+}
+
+export const OrderProducts: FC<OrderProductsProps> = ({
+  products,
+  orderId
+}) => {
+  if (!orderId) {
+    return null
+  }
+
+  console.log('✌️products --->', products)
+
+  return (
+    <Box
+      sx={{
+        gridColumn: { xs: 'span 12', md: 'span 8' }
+      }}
+    >
+      <Box
+        sx={{
+          mb: 3,
+          bgcolor: 'background.default',
+          borderRadius: 2,
+          maxHeight: 612,
+          minHeight: 612,
+          overflow: 'hidden'
+        }}
+      >
+        <Box
+          sx={{
+            overflow: 'auto',
+            maxHeight: 612
+            // p: '30px 20px'
+          }}
+        >
+          <Box
+            display="flex"
+            justifyContent="space-between"
+            alignItems="flex-end"
+            p={2}
+          >
+            <Box>
+              <Typography variant="body1" fontWeight={600} fontSize={22}>
+                Замовлення #{orderId.id}{' '}
+                <span style={{ fontWeight: 400, fontSize: 16 }}>
+                  ({orderId.ordered_products}{' '}
+                  {getProductLabel(orderId.ordered_products)})
+                </span>
+              </Typography>
+            </Box>
+            <Box>
+              <Link
+                to={'/review'}
+                style={{
+                  fontFamily: 'Open Sans',
+                  fontSize: 18,
+                  position: 'relative',
+                  marginRight: 10
+                }}
+              >
+                Залишити відгук
+                <span
+                  style={{
+                    position: 'absolute',
+                    left: 0,
+                    right: 0,
+                    bottom: 0,
+                    height: '1px',
+                    backgroundColor: 'currentColor'
+                  }}
+                />
+              </Link>
+            </Box>
+          </Box>
+          {products.map((product, index) => {
+            const indexColon = product.products.description.indexOf(':')
+            const descriptionToShow =
+              indexColon !== -1
+                ? product.products.description.substring(0, indexColon + 1)
+                : product.products.description
+
+            const firstImage =
+              product.products.images && product.products.images.length > 0
+                ? product.products.images[0].image_url
+                : null
+            const sumProductPrice = product.prices.price * product.quantity
+            const sumProductGrams = product.prices.quantity * product.quantity
```

Comment:
you can create helper functions for this actions and this will reduce complexity of the component